### PR TITLE
feat(Search): Add text and categories search

### PR DESCRIPTION
**Pull request by vogelino**
This PR adds a text and category search into the sidebar and URL. 
It should now be possible to:
- [x] Filter items by text
- [x] Filter items by category
- [x] Combine filters together to find specific items
- [x] Get the same results when reloading the page and using the browser history
- [x] Clear the text search easily
- [x] See the first two categories by default when no parameters exist in the URL
- [x] Cache the result once loaded so it isn't requested from the API a second time
- [x] Avoid changing the filters while a request to the API is happening
- [x] Avoid text searches longer than 100 chars long (This limit can be changed)
- [x] Show the Sidebar button only when opened (Now only says "Suche" and doesn't show the number of active filters, as the categories are always active by default)

New texts key added in Grist:
- `textSearchLabel`
- `textSearchPlaceholder`
- `textSearchCategorySelfHelp`
- `textSearchCategoryAdvising`
- `textSearchCategoryClinics`
- `textSearchCategoryOnlineOffers`
- `textSearchCategoryDisctrictOfficeHelp`
- `optionalFurtherSearchIntroText`
- `filtersButtonLoading`

Changed texts key in Grist:
- `filtersSearchTargetLabel`
- `filterLabel`

Todos:
- [x] URL Params Parsing
- [x] API search
- [x] Button count update
- [x] Refactor complex functions
- [x] Show current search config in list
- [x] Show "No results UI" in list
- [x] Advanced Text Field UI with confirmed state 
- [x] Limit and sanitize text seach query

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205209947864758

### CODE REVIEW

**Comment by vogelino** [comment]
@dnsos & @Esshahn The branch is ready for review. There were many changes so please help me to thoroughly test the new feature. Thx! Let me know if you would change or improve anything. Merci!

**Comment by vogelino** [comment]
> @dnsos & @Esshahn The branch is ready for review. There were many changes so please help me to thoroughly test the new feature. Thx! Let me know if you would change or improve anything. Merci!

There is currently a bug on mobile devices that lead to a full redender of the components when changing filters. I'm looking into it 👀

**Comment by vogelino** [comment]
@Esshahn & @dnsos The bug has been fixed and everything should work fine now. Ready to review!

File: pages/sofortige-hilfe.tsx [hunk 1]
```diff
@@ -43,6 +43,7 @@ const Home: NextPage = () => {
         ) as keyof typeof texts
       ]
 
+  console.log(query)
```
Comment (dnsos) [inline pages/sofortige-hilfe.tsx:46]: That log was hiding well. Probably there is no reason to keep it, right?

Comment (vogelino) [reply pages/sofortige-hilfe.tsx:46]: Good catch!

File: src/lib/hooks/useFilteredFacilitiesCount.ts [hunk 2]
```diff
@@ -23,25 +31,35 @@ export const useFilteredFacilitiesCount = (
...
+    }
 
     const filteredFacilities = facilitiesWithOnlyLabels.filter(
-      (recordLabels) => {
-        if (!isFilteredByTopic && !isFilteredByTarget) return true
-        return isFaclilityActive({
-          isFilteredByTopic,
-          isFilteredByTarget,
+      ([id, recordLabels]) =>
+        isFaclilityActive({
```
Comment (raphael-arce) [inline src/lib/hooks/useFilteredFacilitiesCount.ts:47]: typo: `isFaclilityActive` -> `isFacilityActive`

**Review by raphael-arce — CHANGES_REQUESTED** [review]
Minor: a few typos to correct, otherwise LGTM!

**Review by raphael-arce — CHANGES_REQUESTED** [review]
Great improvements, minor typos other LGTM

**Pull request merged**